src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,373 @@
+            _semanticFactsService = semanticFactsService;
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, _, cancellationToken) = context;
+            var calleeMethodInvocationNode = await context.TryGetRelevantNodeAsync<TInvocationSyntax>().ConfigureAwait(false);
+            if (calleeMethodInvocationNode == null)
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var calleeMethodSymbol = semanticModel.GetSymbolInfo(calleeMethodInvocationNode, cancellationToken).GetAnySymbol() as IMethodSymbol;
+            if (calleeMethodSymbol == null)
+            {
+                return;
+            }
+
+            var isOrdinaryOrExtensionMethod = calleeMethodSymbol.IsOrdinaryMethod() || calleeMethodSymbol.IsExtensionMethod;
+            if (calleeMethodSymbol.DeclaredAccessibility != Accessibility.Private || !isOrdinaryOrExtensionMethod)
+            {
+                return;
+            }
+
+            var calleeMethodDeclarationSyntaxReferences = calleeMethodSymbol.DeclaringSyntaxReferences;
+            if (calleeMethodDeclarationSyntaxReferences.Length != 1)
+            {
+                return;
+            }
+
+            var calleeMethodDeclarationSyntaxReference = calleeMethodDeclarationSyntaxReferences[0];
+            // For C# it will return MethodDeclarationSyntax,
+            // For VB it will return MethodStatementSyntax, and what it is expecting is MethodBlockSyntax
+            var calleeMethodDeclarationOrStatementNode = await calleeMethodDeclarationSyntaxReference.GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            var calleeMethodNode = calleeMethodDeclarationOrStatementNode as TMethodDeclarationSyntax
+                ?? calleeMethodDeclarationOrStatementNode.Parent as TMethodDeclarationSyntax;
+
+            if (calleeMethodNode == null)
+            {
+                return;
+            }
+
+            var inlineExpression = GetInlineExpression(calleeMethodNode);
+            if (_syntaxFacts.IsAwaitExpression(inlineExpression))
+            {
+                // This will make sure there is no duplicate 'await'
+                // Example:
+                // Before:
+                // async Task Caller() => await Callee();
+                // async Task Callee() => await Task.CompletedTask;
+                // After:
+                // async Task Caller() => await Task.CompletedTask;
+                // async Task Callee() => await Task.CompletedTask;
+                // The original inline expression in callee will be 'await Task.CompletedTask'
+                // The caller just need 'Task.CompletedTask' without the 'await'
+                inlineExpression = _syntaxFacts.GetExpressionOfAwaitExpression(inlineExpression) as TExpressionSyntax;
```

Comment:
this is def an interesting case.  do you test all 4 permutations here?  i.e. `caller-no-await, callee-no-await` `caller-await, callee-no-await`, `caller-no-await, callee-await` `caller-await, callee-await`.?

Reply:
Yes,
Case 1: Both caller & callee doesn't have await (This not a problem)
Case 2: If caller doesn't have await, callee has. It means Callee() is fired and forgot/assigned to a variable
`void Caller { var t = Callee()}`
`async Task Callee() { await SomeTask(); }`
The type of `t` is Task,  is expecting `var t = SomeTask();` The await can be safely removed.
Case 3: If Caller has await, callee doesn't,  this not a problem. 
Case 4:  If Caller & callee both have await, then it becomes the example I listed in the comments.
I can update the comments to cover all those cases